src/lib/app.ts

```diff
@@ -67,37 +121,85 @@ export class App {
     this.client = client
   }
 
-  async listApps(org_slug: ListAppRequest): Promise<ListAppResponse> {
+  async listApps(
+    org_slug: ListAppRequest
+  ): Promise<APIResponse<ListAppResponse>> {
     const path = `apps?org_slug=${org_slug}`
-    return await this.client.restOrThrow(path)
+    return await this.client.safeRest(path)
   }
 
-  async getApp(app_name: GetAppRequest): Promise<AppResponse> {
+  async listAppsDetailed(
+    org_slug: ListAppRequest
+  ): Promise<APIResponse<ListAppDetailedResponse>> {
+    const response = await this.client.safeGqlPost<
+      string,
+      { organization: any }
+    >({
+      query: getOrganizationAppsDetailedQuery,
+      variables: { slug: org_slug },
+    })
+    if (response.error) {
+      return response
+    }
+    return {
+      data: parseOrgResponse(response.data.organization).apps,
+      error: undefined,
+    }
+  }
+
+  async getApp(app_name: GetAppRequest): Promise<APIResponse<AppResponse>> {
     const path = `apps/${app_name}`
-    return await this.client.restOrThrow(path)
+    return await this.client.safeRest(path)
   }
 
-  async getAppDetailed(app_name: GetAppRequest): Promise<AppResponse> {
-    const { app } = await this.client.gqlPostOrThrow({
-      query: getAppQuery,
+  async getAppDetailed(
+    app_name: GetAppRequest
+  ): Promise<APIResponse<AppDetailedResponse>> {
+    const response = await this.client.safeGqlPost<string, { app: any }>({
```

Comment:
Can we avoid typing `app: any` in calls to `safeGqlPost`? Always define the interface types returned from graphql api call.